<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="bcr_arm">
<!-- Link definitions -->
  <link name="base">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
   
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
    </collision>
  </link>
 <link name="base_link">
    <visual>
      <origin xyz="0.0 0.0 -0.05" rpy="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="0.1" radius="0.5"/>
      </geometry>
   
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
    </collision>
  </link>

  <link name="link1">
    <visual>
       <origin xyz="0.0 0.0 0.5" rpy="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="1.0" radius="0.05"/>
      </geometry>
   
    </visual>
    <collision>
      <geometry>
        <cylinder length="1.0" radius="0.05"/>
      </geometry>
    </collision>
  </link>

  <link name="link2">
    <visual>
      <origin xyz="0.0 0.0 0.5" rpy="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="1.0" radius="0.025"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="1.0" radius="0.025"/>
      </geometry>
    </collision>
  </link>

  <link name="end_effector">
    <visual>
      <geometry>
        <box size="0.025 0.25 0.025"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.1"/>
      </geometry>
    </collision>
  </link>

  <!-- Joint definitions -->
  <joint name="joint1" type="revolute">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="base"/>
    <child link="link1"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="0.1"/>
  </joint>
  <joint name="joint" type="revolute">
    <origin xyz="0 0 0.0" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
    <axis xyz="0 0 1"/>
    <limit lower="-3.14" upper="3.14" effort="100" velocity="0.1"/>
  </joint>


  <joint name="joint2" type="revolute">
    <origin xyz="0 0 1.0" rpy="0 0 0"/>
    <parent link="link1"/>
    <child link="link2"/>
    <axis xyz="0 1 0"/>
    <limit lower="-2.5" upper="2.5" effort="100" velocity="0.1"/>
  </joint>

  <joint name="joint3" type="revolute">
    <origin xyz="0 0 1.0" rpy="1.57 0 1.57"/>
    <parent link="link2"/>
    <child link="end_effector"/>
    <axis xyz="1 0 0"/>
    <limit lower="-2.5" upper="2.5" effort="100" velocity="0.1"/>
  </joint>
</robot>

